src/pages/PendingRequests/PendingRequests.tsx

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useMemo, useState } from "react";
+import { Container, Row, Col, Button } from "react-bootstrap";
+import Table from "components/Table/Table";
+import { REQUEST_COLUMNS } from "./RequestColumns";
+import { IPendingTable } from "./util";
+import axios from "axios";
+import { Modal } from "react-bootstrap";
+
+const PendingRequests = () => {
+  const [showCurrent, setCurrentData] = useState(true);
+  const [tableData, setTableData] = useState<Record<string, IPendingTable>[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [showModal, setShowModal] = useState(false);
+  const [selectedRow, setSelectedRow] = useState<{ id: any } | null>({ id: null });
+  const [action, setAction] = useState<"accept" | "reject" | null>(null);
+
+  useEffect(() => {
+    fetchData();
+  }, [showCurrent]);
+
+  const fetchData = async () => {
+    let endpoint = showCurrent ? "pending" : "processed";
+    try {
+      setLoading(true);
+
+      const response = await axios.get(`http://127.0.0.1:3000/api/v1/account_requests/${endpoint}`); // Make a GET request to the 'pending_requests' endpoint
```

Comment:
Using the useAPI hook is a more efficient approach as compared to creating a new Axios instance. Additionally, it promotes consistency in the code throughout the application.